### PR TITLE
feat(SpokePool): Remove maxCount related logic

### PR DESCRIPTION
**Pull request by nicholaspai**
Supporting maxcount adds extra gas cost to soon-to-be deprecated deposit and fill functions. This PR removes them and also removes related functions like `requestRepaymentChainId`.

A maxCount is likely to be added in future to new deposit and fill functions (i.e. USS-functions) but we'll add them in a follow-up PR

### CODE REVIEW

File: contracts/SpokePool.sol [hunk 1]
```diff
@@ -447,7 +454,7 @@ abstract contract SpokePool is
         int64 relayerFeePct,
         uint32 quoteTimestamp,
         bytes memory message,
-        uint256 maxCount
+        uint256 // maxCount. Deprecated.
```
Comment (james-a-morris) [inline contracts/SpokePool.sol:457]: Should we still keep this as a named parameter - even if we're ignoring it?

**Review by james-a-morris — COMMENTED** [review]
Left a comment about signatures. I will follow up with more comments in the AM

File: contracts/SpokePool.sol (hunk 1 above)
Comment (nicholaspai) [reply contracts/SpokePool.sol:457]: that would trigger the linter since its unused, why keep it named?

**Review by pxrl — COMMENTED** [review]
No blockers from me but I'm interested to see what other comments James might have.

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**